graph/is_graph_bipartite.cpp

```diff
@@ -0,0 +1,162 @@
+/**
+ *  @file is_graph_bipartite
+ *
+ *	@brief Algorithm to check whether a graph is [bipartite](https://en.wikipedia.org/wiki/Bipartite_graph)
+ *	
+ *	@details
+ *	A graph is a collection of nodes also called vertices and these vertices
+ *	are connected by edges.A bipartite graph is a graph whose vertices can be
+ * 	divided into two disjoint and independent sets U and V such that every edge 
+ *	connects a vertex in U to one in V. 	
+ *	
+ *	The given Algorithm will determine whether the given graph is bipartite or not
+ *	
+ *	
+ * 	Example - Here is a graph g1 with 5 vertices and is bipartite
+ *	
+ *		1   4
+ *	   / \ / \
+ *	  2   3   5
+ *	
+ *	Example - Here is a graph G2 with 3 vertices and is not bipartite
+ *	
+ *		1 --- 2
+ *		 \   /
+ *		   3
+ *	
+ *	@author [Akshat Vaya](https://github.com/AkVaya)
+ *		
+ */
+#include <iostream>
+#include <vector>
+#include <queue>
+
+/**
+ * Class for representing graph as an adjacency list.
+ */
+/**
+ *	@namespace graph
+ *  @brief Graph algorithms
+ */
+namespace graph{
+	/**
+	 * @namespace is_graph_bipartite
+	 * @brief Functions for checking whether a graph is bipartite or not
+	 */
+	namespace is_graph_bipartite{
+
+		class Graph {
+ 			private: 
+ 				int n;						/// size of the graph
+
+	    		std::vector<std::vector <int> > adj;	/// adj stores the graph as an adjacency list
+
+		    	std::vector<int> side;			///stores the side of the vertex
+
+				static const int nax = 5e5 + 1;
+
+
+	 		public:
+    		/**
+    		 * @brief Constructor that initializes the graph on creation
+    		 */
```

Comment:
This is also missing parameter documentation.
```suggestion
    		        /**
    		        * @brief Constructor that initializes the graph on creation
    		        */
```